From 9f8e1ddf11d61b7a1eb57aa7e80dc007427d8cac Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Tue, 13 Nov 2018 20:18:41 -0700
Subject: [PATCH 8/8] Allow mediaserver to read device directories

* Needed both to allow ExternalCameraProvider to probe for
  video devices as well as allow access/recognition of
  external storage

* avc: denied { read } for name="/" dev="tmpfs" ino=6954
  scontext=u:r:mediaserver:s0 tcontext=u:object_r:device:s0
  tclass=dir permissive=1
* avc: denied { open } for name="/" dev="tmpfs" ino=6954
  scontext=u:r:mediaserver:s0 tcontext=u:object_r:device:s0
  tclass=dir permissive=1

Change-Id: I32540b8ec48ff61568094ec87d1bad64570a8e55
---
 public/mediaserver.te | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/public/mediaserver.te b/public/mediaserver.te
index 400ebcc8..ae9718f1 100644
--- a/public/mediaserver.te
+++ b/public/mediaserver.te
@@ -136,6 +136,11 @@ allow mediaserver media_rw_data_file:file create_file_perms;
 # Access to media in /data/preloads
 allow mediaserver preloads_media_file:file { getattr read ioctl };
 
+ifelse(target_has_legacy_camera_hal1, `true',
+  allow mediaserver device:dir r_dir_perms;
+,
+)
+
 allow mediaserver ion_device:chr_file r_file_perms;
 allow mediaserver hal_graphics_allocator:fd use;
 allow mediaserver hal_graphics_composer:fd use;
-- 
2.17.1

